perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -20,31 +20,113 @@
+        'blocksize': '512k'
+    },
+    'random_write': {
+        'name': 'random_write',
+        'rwkind': 'randwrite',
+        'blocksize': '4k'
+    },
+    'random_read': {
+        'name': 'random_read',
+        'rwkind': 'randread',
+        'blocksize': '4k'
+    }
+}
+
 
 FLAGS = flags.FLAGS
 
-flags.DEFINE_string('fio_benchmark_filename', 'fio_benchmark_file',
-                    'scratch file that fio will use')
-flags.DEFINE_string('fio_jobfile', 'fio.job', 'job file that fio will use')
-flags.DEFINE_integer('memory_multiple', 10,
-                     'size of fio scratch file compared to main memory size.')
+flags.DEFINE_string('fio_jobfile', None,
+                    'Job file that fio will use. If not given, use a job file '
+                    'bundled with PKB. Cannot use with --generate_scenarios.')
+flags.DEFINE_list('generate_scenarios', None,
+                  'Generate a job file with the given scenarios. Special '
+                  'scenario \'all\' generates all scenarios. Available '
+                  'scenarios are sequential_write, sequential_read, '
+                  'random_write, and random_read. Cannot use with '
+                  '--fio_jobfile.')
 flags.DEFINE_boolean('against_device', False,
-                     'Test direct against scratch disk. If set True, will '
-                     'create a modified job file in temp directory, '
-                     'which ignores directory and filename parameter.')
+                     'Unmount the device\'s filesystem so we can test against '
+                     'the raw block device. If --generate-scenarios is given, '
+                     'will generate a job file that uses the block device.')
 flags.DEFINE_string('device_fill_size', '100%',
                     'The amount of device to fill in prepare stage. '
-                    'This flag is only valid when against_device=True. '
                     'The valid value can either be an integer, which '
                     'represents the number of bytes to fill or a '
                     'percentage, which represents the percentage '
-                    'of the device.')
+                    'of the device. Default to filling 100% of a raw device or '
+                    '90% of a filesystem.')
+flags.DEFINE_string('io_depths', '1',
+                    'IO queue depths to run on. Can specify a single number, '
+                    'like --io_depths=1, or a range, like --io_depths=1-4')
+flags.DEFINE_integer('working_set_size', None,
+                     'The size of the working set, in GB. If not given, use '
+                     'the full size of the device.',
+                     lower_bound=0)
+
+
+
+FLAGS_IGNORED_FOR_CUSTOM_JOBFILE = {
+    'generate_scenarios', 'io_depths'}
+
+
+IODEPTHS_REGEXP = re.compile(r'(\d+)(-(\d+))?')
+
+
+def IODepthsValidator(string):
+  match = IODEPTHS_REGEXP.match(string)
+  return match and match.end() == len(string) and int(match.group(1)) > 0
```

Comment:
Nit: adding `$` to the regex above will be equivalent to `match.end() == len(string)`.